app/controllers/action_officers_controller.rb

```diff
@@ -1,68 +1,84 @@
 class ActionOfficersController < ApplicationController
   before_action :authenticate_user!, PQUserFilter
-  before_action :set_action_officer, only: [:show, :edit, :update, :destroy]
-  before_action :prepare_dropdowns
 
   def index
-    @action_officers = ActionOfficer.all.joins(:deputy_director => :division).order('lower(divisions.name)').order('lower(action_officers.name)')
+    @action_officers = 
+      ActionOfficer.all
+        .joins(:deputy_director => :division)
+        .order('lower(divisions.name)')
+        .order('lower(action_officers.name)')
   end
 
-  def new
-    @action_officer = ActionOfficer.new
+  def show
+    loading_records(:existing) 
   end
 
-  def find
-    @results = ActionOfficer.by_name(params[:q]).select(:id, :name)
-    render json: @results
+  def new
+    loading_records(:new)
   end
 
   def create
-    begin
-    @action_officer = ActionOfficer.new(action_officer_params)
-    if @action_officer.save
-      flash[:success] = 'Action officer was successfully created.'
-      redirect_to action_officers_path
-      return
+    loading_records(:new) do      
+      if @action_officer.update(action_officer_params)
+        flash[:success] = 'Action officer was successfully created'
+        redirect_to action_officers_path
+      else
+        flash[:error] = 'Action officer could not be created'
+        render action: 'new'
+      end
     end
-    rescue => err
-      @action_officer.errors[:base] << handle_db_error(err)
-    end
-    render action: 'new'
+  end
+
+  def edit
+   loading_records(:existing)
   end
 
   def update
-    begin
-    if @action_officer.update(action_officer_params)
-      flash[:success] = 'Action officer was successfully updated.'
-      redirect_to @action_officer 
-      return
+    loading_records(:existing) do
+      if @action_officer.update(action_officer_params)
+        flash[:success] = 'Action officer was successfully updated'
+        redirect_to @action_officer 
```

Comment:
Maybe it would be better to be explicit here instead of relying on Rails' magical conventions. 

     redirect_to action_officer_path(@action_officer.id)